test/http_header_end2end_test.go

```diff
@@ -0,0 +1,261 @@
+/*
+*
+* Copyright 2022 gRPC authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+ */
+package test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal/transport"
+	"google.golang.org/grpc/status"
+	testpb "google.golang.org/grpc/test/grpc_testing"
+)
+
+func (s) TestHTTPHeaderFrameErrorHandlingHTTPMode(t *testing.T) {
+
+	type test struct {
+		name    string
+		header  []string
+		errCode codes.Code
+	}
+
+	var tests []test
+
+	// Non-gRPC content-type fallback path.
+	for httpCode := range transport.HTTPStatusConvTab {
+		tests = append(tests, test{
+			name: fmt.Sprintf("Non-gRPC content-type fallback path with httpCode: %v", httpCode),
+			header: []string{
+				":status", fmt.Sprintf("%d", httpCode),
+				"content-type", "text/html", // non-gRPC content type to switch to HTTP mode.
+				"grpc-status", "1", // Make up a gRPC status error
+				"grpc-status-details-bin", "???", // Make up a gRPC field parsing error
+			},
+			errCode: transport.HTTPStatusConvTab[int(httpCode)],
+		})
+	}
+
+	// Missing content-type fallback path.
+	for httpCode := range transport.HTTPStatusConvTab {
+		tests = append(tests, test{
+			name: fmt.Sprintf("Missing content-type fallback path with httpCode: %v", httpCode),
+			header: []string{
+				":status", fmt.Sprintf("%d", httpCode),
+				// Omitting content type to switch to HTTP mode.
+				"grpc-status", "1", // Make up a gRPC status error
+				"grpc-status-details-bin", "???", // Make up a gRPC field parsing error
+			},
+			errCode: transport.HTTPStatusConvTab[int(httpCode)],
+		})
+	}
+
+	// Malformed HTTP status when fallback.
+	tests = append(tests, test{
+		name: "Malformed HTTP status when fallback",
+		header: []string{
+			":status", "abc",
+			// Omitting content type to switch to HTTP mode.
+			"grpc-status", "1", // Make up a gRPC status error
+			"grpc-status-details-bin", "???", // Make up a gRPC field parsing error
+		},
+		errCode: codes.Internal,
+	})
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			listAddr, cleanup, err := startServer(t, test.header)
```

Comment:
Nit: Use `lisAddr` or `serverAddr` instead of `listAddr`.

Same applies to other tests where `listAddr` is used.

Reply:
done